packages/dashboard-frontend/src/components/EditorTools/index.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { Button, Divider, Flex, FlexItem } from '@patternfly/react-core';
+import { CompressIcon, CopyIcon, DownloadIcon, ExpandIcon } from '@patternfly/react-icons';
+import React from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import stringify from '../../services/helpers/editor';
+import styles from './index.module.css';
+import devfileApi from '../../services/devfileApi';
+
+type Props = {
+  content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile;
+  handleExpand: (isExpand: boolean) => void;
+};
+
+type State = {
+  copied?: boolean;
+  isExpanded: boolean;
+  contentText: string;
+  isDevWorkspace: boolean;
+  contentBlobUrl: string;
+};
+
+class EditorTools extends React.PureComponent<Props, State> {
+  private copiedTimer: number | undefined;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isExpanded: false,
+      contentText: '',
+      isDevWorkspace: false,
+      contentBlobUrl: '',
+    };
+  }
+
+  public componentDidMount(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      const isDevWorkspace = this.isDevWorkspace(content);
+      const contentBlobUrl = URL.createObjectURL(
+        new Blob([contentText], { type: 'application/x-yaml' }),
+      );
+      this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  public componentDidUpdate(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      if (contentText !== this.state.contentText) {
+        const isDevWorkspace = this.isDevWorkspace(content);
+        const contentBlobUrl = URL.createObjectURL(
+          new Blob([contentText], { type: 'application/x-yaml' }),
+        );
+        this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  private handleExpand(): void {
+    if (this.state.isExpanded) {
+      window.postMessage('show-navbar', '*');
```

Comment:
I'd ask you to use the `ToggleBarContext` component to switch the navbar visibility. Since the dashboard stopped opening editors in the iframe, we don't have to listen to messages from iframes and send messages as well.

Here is how to use the context for toggling bars: https://github.com/eclipse-che/che-dashboard/blob/36eb9624ebb2f0f212090a0d48a418be11f8e98a/packages/dashboard-frontend/src/containers/IdeLoader/index.tsx

Reply:
Ok. I have fixed it.